migrations/V193__Add_MTO_Category_Delete_Trigger.sql

```diff
@@ -0,0 +1,89 @@
+/*
+    This SQL script sets up a trigger and a supporting function to handle custom cascading
+    deletion behavior for categories in the `mto_category` table, per the application requirements.
+
+    Updated Assumption (per user's note):
+    -------------------------------------
+    Instead of assigning milestones to a designated "Uncategorized" category with a known UUID,
+    we will set their `mto_category_id` to NULL (a "nil" UUID), indicating they are now uncategorized.
+
+    Requirements:
+    - Deleting a subcategory:
+        * All milestones referencing that subcategory should be reassigned to its parent category.
+    - Deleting a top-level category:
+        * All milestones referencing the deleted top-level category or its direct subcategories
+          should have `mto_category_id` set to NULL, indicating they are now uncategorized.
+        * All direct subcategories of the deleted category should be deleted.
+
+    Why the "CASCADE" Keyword Isn't Used:
+    -------------------------------------
+    The standard ON DELETE CASCADE simply removes dependent rows. Here, we need to reassign
+    references rather than just delete them. Since this behavior is more complex, we implement
+    custom logic in a trigger and a PL/pgSQL function rather than relying on cascade deletes.
+
+    Usage:
+    - Once this script is applied, a `DELETE FROM mto_category WHERE id = 'some-category-uuid';`
+      will trigger the logic to reassign milestones and remove subcategories as defined.
+
+    Assumptions:
+    - When deleting a top-level category, milestones are "uncategorized" by setting their
+      `mto_category_id` to NULL.
+    - For subcategories, milestones are moved up to the parent category.
+    - This handles one level of subcategories. If a deeper hierarchy is required,
+      the logic should be extended to recursively handle all descendants.
+*/
+
+-- Drop existing trigger and function to allow clean re-creation
+DROP TRIGGER IF EXISTS mto_category_before_delete ON mto_category;
+DROP FUNCTION IF EXISTS rebalance_milestones_before_category_delete();
+
+-- Create the trigger function that implements the custom cascading logic
+CREATE OR REPLACE FUNCTION rebalance_milestones_before_category_delete()
+  RETURNS TRIGGER AS $$
+DECLARE
+  cat_model_plan_id UUID;
+  is_top_level BOOLEAN;
+BEGIN
+  -- Determine if the category is top-level or a subcategory

```

Comment:
Questions, does it matter if it is a parent category or not? Couldn't you always use the parent category id field? (for a top level field, that is null, so it would still correctly set it to null). That might simplify the logic. EG you always just set the milestone_id to the category parent_id

Reply:
eg, something like
```sql
BEGIN
    -- Update the milestones to reference the parent category or null
    UPDATE mto_milestone
    SET mto_category_id = OLD.parent_id
    WHERE mto_category_id = OLD.id;

    RETURN OLD;
END;
```